music21/musicxml/m21ToXml.py

```diff
@@ -2554,16 +2596,18 @@ def instrumentSetup(self):
             self.midiChannelList.append(thisInstrument.midiChannel)
             # environLocal.printDebug(['midiChannel list', self.midiChannelList])
 
-            # add to list for checking on next part
-            if self.parent is not None:
-                self.parent.instrumentList.append(thisInstrument)
-            # force this instrument into this part
-            # meterStream is only used here if there are no measures
-            # defined in this part
-
-            if thisInstrument.instrumentId is None:
+            # Enforce uniqueness
+            if (thisInstrument.instrumentId is None
+                or (self.parent
+                    and thisInstrument.instrumentId in self.parent.instrumentIdList)):
```

Comment:
maybe here's the place to check for numerical ids.

Reply:
My bad, but I didn't follow this comment.